pkg/endpoint/endpoint.go

```diff
@@ -301,8 +302,7 @@ type Endpoint struct {
 	// realizedRedirects maps the ID of each proxy redirect that has been
 	// successfully added into a proxy for this endpoint, to the redirect's
 	// proxy port number.
-	// You must hold Endpoint.mutex to read or write it.
-	realizedRedirects map[string]uint16
+	realizedRedirects sync.Map
```

Comment:
I don't fully understand the semantics of sync.Map, I know it's only advantageous in certain access patterns.  Could we add a comment explaining the specific case of endpoints and why this an advantage in this case?

Reply:
It's good you asked this question.

As part of writing out the access patterns, I realized we could do this in a much cleaner way, just with some extra comments around locking. So, this PR will change a bit. Thanks for that!